hollow/src/main/java/com/netflix/hollow/api/producer/fs/HollowFilesystemBlobStorageCleaner.java

```diff
@@ -0,0 +1,92 @@
+/*
+ *
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.hollow.api.producer.fs;
+
+import com.netflix.hollow.api.producer.HollowProducer;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.logging.Logger;
+
+public class HollowFilesystemBlobStorageCleaner extends HollowProducer.BlobStorageCleaner {
+
+    private final Logger log = Logger.getLogger(HollowFilesystemBlobStorageCleaner.class.getName());
+
+    private int numOfSnapshotsToKeep;
+    private final File blobStoreDir;
+
+    public HollowFilesystemBlobStorageCleaner(File blobStoreDir) {
+        this(blobStoreDir,5);
+    }
+
+    public HollowFilesystemBlobStorageCleaner(File blobStoreDir, int numOfSnapshotsToKeep) {
+        this.blobStoreDir = blobStoreDir;
+        this.numOfSnapshotsToKeep = numOfSnapshotsToKeep;
+    }
+
+    /**
+     * Cleans snapshot to keep the last 'n' snapshots. Defaults to 5.
+     */
+    @Override
+    public void cleanSnapshots() {
+        File[] files = getFilesByType(HollowProducer.Blob.Type.SNAPSHOT.prefix);
+
+        if(files == null || files.length <= numOfSnapshotsToKeep) {
+            return;
+        }
+
+        sortByLastModified(files);
+
+        File[] filesToRemove = Arrays.copyOfRange(files, 0, numOfSnapshotsToKeep - 1);
+
+        for(File file : filesToRemove) {
+            boolean deleted = file.delete();
+            if(!deleted) {
+                log.warning("Could not delete snapshot " + file.getPath());
+            }
+        }
+    }
+
+
+    @Override
+    public void cleanDeltas() { }
+
+    @Override
+    public void cleanReverseDeltas() { }
+
+    private void sortByLastModified(File[] files) {
+        Arrays.sort(files, new Comparator<File>() {
+            public int compare(File f1, File f2) {
+                return Long.compare(f1.lastModified(), f2.lastModified());
```

Comment:
Remove the reversal sort order. It seems that the most recent files will be cleaned up first. The oldest file should be the first to be cleaned up.

Reply:
Tackled in another way:  modified the for loop to always keep the latest 'n' files -> https://github.com/Netflix/hollow/pull/94/files#diff-4dbcd2ad1aede1e6ad3380cca689a0a0R58

Also modified the test to verify this -> https://github.com/Netflix/hollow/pull/94/files#diff-b7f49725b4f12dbaf20d35b021c1fea2R91. It uses a custom `VersionMinter` to increase an `AtomicInteger` so in the test we always guarantee the name of the snapshots in comparison with `VersionMinterWithCounter`.

Reply:
Love the custom version minter for unit test.

Since you're keeping the reverse sort order, you can avoid sorting twice by altering the custom comparator slightly to `compare(f2.lastModified(), f1.lastModified());`, i.e. swap the order from (f1,f2) to (f2,f1).

It's a subtle difference, but in my experience relatively idiomatic.